### PR TITLE
Add Clermontyping 

### PR DESCRIPTION
**Pull request by happykhan**
Adding recipe for Clermontyping tool. 

---

### CODE REVIEW

File: recipes/clermontyping/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+cp $SRC_DIR/clermonTyping.sh ${PREFIX}
```
Comment (martin-g) [inline recipes/clermontyping/build.sh:1]: This should probably be copied to ${PREFIX}/bin ?!

Comment (martin-g) [inline recipes/clermontyping/build.sh:1]:
```suggestion
#!/usr/bin/env bash

set -xe

cp $SRC_DIR/clermonTyping.sh ${PREFIX}
```

File: recipes/clermontyping/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package:
+  name: "clermontyping"
+  version: "1.0.1"
+  
+source:
+  url: "https://github.com/happykhan/ClermonTyping/archive/refs/tags/1.0.2.tar.gz"
```
Comment (martin-g) [inline recipes/clermontyping/meta.yaml:6]: 1.0.1 or 1.0.2 ?

File: recipes/clermontyping/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  sha256: 6b14a7a1450f13dfc5fc4ca39349baf94d9a09b726f982213119284d06bf2f8c
+
+build:
+  number: 0
+  noarch: generic
+  run_exports:
+    - {{ pin_subpackage('clermontyping', max_pin="1.0.1") }}
```
Comment (martin-g) [inline recipes/clermontyping/meta.yaml:13]:
```suggestion
    - {{ pin_subpackage('clermontyping', max_pin="x") }}  
```

File: recipes/clermontyping/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - mash
+
+test:
+  commands:
+    - clermonTyping.sh -h
+
+about:
+  home: "https://github.com/happykhan/ClermonTyping"
+  summary: " determine the Clermont  E coli phylotype"
+  license: GPLv3
```
Comment (martin-g) [inline recipes/clermontyping/meta.yaml:39]:
```suggestion
  license: GPL-3.0-only
```

**Comment by happykhan** [comment]
Let's see how this goes with those changes. On my local build it was failing on the mulled test

```
13:15:45 BIOCONDA INFO (ERR) Passing program arguments as a string may be a security hazard if combined with untrusted input
13:15:45 BIOCONDA INFO (ERR) [Jun 28 13:15:45] DEBU Run file [/Users/nfareed/micromamba/envs/bioconda/lib/python3.9/site-packages/galaxy/tools/deps/mulled/invfile.lua]
13:15:45 BIOCONDA INFO (ERR) [Jun 28 13:15:45] INFO Invoke Task [build]
13:15:45 BIOCONDA INFO (ERR) [Jun 28 13:15:45] STEP Run image [quay.io/bioconda/create-env:latest] with command [[rm -rf /data/dist]]
13:15:45 BIOCONDA INFO (ERR) [Jun 28 13:15:45] DEBU Creating container [step-15fa0bf807]
13:15:45 BIOCONDA INFO (ERR) [Jun 28 13:15:45] DEBU Created container [926cb12a9d9d step-15fa0bf807], starting it
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Container [926cb12a9d9d step-15fa0bf807] started, waiting for completion
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Container [926cb12a9d9d step-15fa0bf807] completed with exit code [0] as expected
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Container [926cb12a9d9d step-15fa0bf807] removed
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] STEP Run image [quay.io/bioconda/create-env:latest] with command [[/bin/sh -c conda() { mamba "${@}" ; } && conda install  -c conda-forge -c file:///Users/nfareed/micromamba/envs/bioconda/conda-bld -c bioconda -c defaults  clermontyping=1.0.1=py312hdfd78af_0 -p /usr/local --copy --yes --quiet&&create-env --conda=: /usr/local]]
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Creating container [step-7645ecc53b]
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Created container [8d0eb9cb71ea step-7645ecc53b], starting it
13:15:46 BIOCONDA INFO (ERR) [Jun 28 13:15:46] DEBU Container [8d0eb9cb71ea step-7645ecc53b] started, waiting for completion
13:15:55 BIOCONDA INFO (ERR) [Jun 28 13:15:55] SOUT Could not solve for environment specs
13:15:55 BIOCONDA INFO (ERR) [Jun 28 13:15:55] SOUT The following package could not be installed
13:15:55 BIOCONDA INFO (ERR) [Jun 28 13:15:55] SOUT └─ clermontyping ==1.0.1 py312hdfd78af_0 is not installable because it requires
13:15:55 BIOCONDA INFO (ERR) [Jun 28 13:15:55] SOUT └─ blast, which does not exist (perhaps a missing channel).
13:15:55 BIOCONDA INFO (ERR) [Jun 28 13:15:55] ERRO Task processing failed: Unexpected exit code [1] of container [8d0eb9cb71ea step-7645ecc53b], container preserved
.13:16:45 BIOCONDA INFO (OUT) .
13:16:45 BIOCONDA ERROR COMMAND FAILED (exited with 1): mulled-build build-and-test clermontyping=1.0.1--py312hdfd78af_0 -n biocontainers --test bash -c '/usr/local/env-execute true && . /usr/local/env-activate.sh && clermonTyping.sh -h' --channels conda-forge,file:///Users/nfareed/micromamba/envs/bioconda/conda-bld,bioconda,defaults --involucro-path /Users/nfareed/micromamba/envs/bioconda/lib/python3.9/site-packages/bioconda_utils/involucro

13:16:45 BIOCONDA ERROR TEST FAILED: recipes/clermontyping
13:16:45 BIOCONDA INFO (COMMAND) conda build purge
13:16:48 BIOCONDA ERROR BUILD SUMMARY: of 1 recipes, 1 failed and 0 were skipped. Details of recipes and environments follow.
13:16:48 BIOCONDA ERROR BUILD SUMMARY: FAILED recipe recipes/clermontyping
```

**Review by martin-g — APPROVED** [review]

**Pull request merged**